libexec/rtld-elf/rtld_c18n.c

```diff
@@ -882,17 +883,23 @@ unwind_cursor(void)
 	 * buffer.
 	 */
 
-	return (get_trusted_stk()->previous);
+	if (C18N_ENABLED)
+		*buf = cheri_seal(get_trusted_stk()->previous,
+		    sealer_trusted_stk);
+	else
+		*buf = NULL;
 }
 
+/*
+ * XXX: These functions are kept here for compatibility with old libunwind.
```

Comment:
libc and libunwind (ditto for the other instance below)

Reply:
And maybe group all the compat interfaces together at the end?

Reply:
It is possible to move all the function definitions to the end, but the declaration of `_rtld_unw_setcontext_impl` has to stay in front because of the function-pointer workaround employed in the previous release. `_rtld_unw_setcontext_impl` has to be wrapped into `_rtld_unw_setcontext_ptr` in `c18n_init2`.